src/inet/TCPEndPointImplLwIP.cpp

```diff
@@ -570,7 +570,7 @@ TCPEndPointImplLwIP::BufferOffset TCPEndPointImplLwIP::FindStartOfUnsent()
     while (leftToSkip > 0)
     {
         VerifyOrDie(!startOfUnsent.buffer.IsNull());
-        uint16_t bufDataLen = startOfUnsent.buffer->DataLength();
+        uint16_t bufDataLen = static_cast<uint16_t>(startOfUnsent.buffer->DataLength());
```

Comment:
This is concerning me, in that a larger than UINT16_MAX `DataLength` would cause issues

Reply:
I agree with the concern. Stumbled on the LwIP uint16 limitation midway. So, what I was thinking is to have a hard check up front on the TCP payload to limit it to less than UINT16_MAX for LwIP based systems so that all subsequent casts are de-risked. Otherwise, it would need a much more extensive change I think.

Reply:
PacketBuffer::New() has a check against UINT16_MAX for LwIP. This should account for buffers created for the outgoing path. For incoming, LwIP itself would only pass a PacketBuffer that is limited to UINT16_MAX. So, the addition of the MaxTCPPayload in the CASE session params would help in informing the peer of the local system limits. https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/8924